filebeat/prospector/reloader.go

```diff
@@ -0,0 +1,167 @@
+package prospector
+
+import (
+	"expvar"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/filebeat/registrar"
+	"github.com/elastic/beats/libbeat/cfgfile"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	debugr            = logp.MakeDebug("filebeat.reloader")
+	configReloads     = expvar.NewInt("filebeat.config.reloads")
+	prospectorStarts  = expvar.NewInt("filebeat.config.prospector.starts")
+	prospectorStops   = expvar.NewInt("filebeat.config.prospector.stops")
+	prospectorRunning = expvar.NewInt("filebeat.config.prospector.running")
+)
+
+type ProspectorReloader struct {
+	registry  *registry
+	config    cfgfile.ReloadConfig
+	outlet    Outlet
+	done      chan struct{}
+	wg        sync.WaitGroup
+	registrar *registrar.Registrar
+}
+
+func NewProspectorReloader(cfg *common.Config, outlet Outlet, registrar *registrar.Registrar) *ProspectorReloader {
+
+	config := cfgfile.DefaultReloadConfig
+	cfg.Unpack(&config)
+
+	return &ProspectorReloader{
+		registry:  newRegistry(),
+		config:    config,
+		outlet:    outlet,
+		done:      make(chan struct{}),
+		registrar: registrar,
+	}
+}
+
+func (r *ProspectorReloader) Run() {
+
+	logp.Info("Prospector reloader started")
+
+	r.wg.Add(1)
+	defer r.wg.Done()
+
+	// Stop all running modules when method finishes
+	defer r.stopProspectors(r.registry.CopyList())
+
+	path := r.config.Path
+	if !filepath.IsAbs(path) {
+		path = filepath.Join(cfgfile.GetPathConfig(), path)
```

Comment:
`GetPathConfig` was around for quite some time I think, or at least the logic around it. I was not aware of `paths.Resolve` that is why I used `GetPathConfig`. I changed it now in the reloaders.